editoast: add get rolling stock by name endpoint

younesschrifi opened the pull request:
Add an enpoint to retrieve a rolling stock given its name.

younesschrifi committed:
editoast: add get rolling stock by name endpoint

leovalais commented on editoast/src/views/rolling_stocks/mod.rs:
```diff
@@ -103,7 +107,7 @@ pub enum RollingStockError {
     CannotCreateCompoundImage,
     #[error("Rolling stock '{rolling_stock_id}' could not be found")]
     #[editoast_error(status = 404)]
-    NotFound { rolling_stock_id: i64 },
```
Either make a second variant or do something like:

```
KeyNotFound { rolling_stock_key: RollingStockKey }

#serde tag = "type"
enum RollingStockKey = Id(i64) | Name(String)
```

because a RS ID is not a string ;)

leovalais commented on editoast/src/views/rolling_stocks/mod.rs:
```diff
@@ -462,13 +498,19 @@ async fn create_livery(
 
-pub async fn retrieve_existing_rolling_stock(
+/// Retrieve a rolling stock by id or by name
+pub async fn retrieve_existing_rolling_stock<T>(
```
If you decide to make an enum `RollingStockKey` you can pass it in parameter here, eliminating the need for generics.

Otherwise I'd suggest naming the key `K` to stay consistent with the definitions of `modelsv2::prelude`.

leovalais commented on editoast/src/views/rolling_stocks/mod.rs:
```diff
@@ -608,14 +651,27 @@ pub mod tests {
+
+        // get by name
+        let req = rolling_stock_get_by_name_request(rolling_stock.name.clone());
+        let response = call_service(&app, req).await;
```
Could you move that into a new test case?

leovalais commented on editoast/src/views/rolling_stocks/mod.rs:
```diff
@@ -608,14 +651,27 @@ pub mod tests {
 
+        assert_eq!(get_response.status(), StatusCode::NOT_FOUND);
+        // get by name
+        let app = create_test_service().await;
```
likewise

leovalais reviewed (changes requested):
Mostly lgtm, just some details:

leovalais reviewed (approved):
Tested, thanks!

younesschrifi committed:
editoast: add light rolling stock endpoint get by name

flomonster reviewed (approved):
LGTM

SharglutDev reviewed (approved):
Lgtm. You can rest now :)

sim51 committed:
front: fix api i18n checker for message variable with object

When the script found an error definition in openapi.yml file, it tries to generate its message with a dummy variable object.
And this object can't be well defined, specially for objects.
This leads to an error for the script, even if it's valid.
